ceremonies/src/lib.rs

```diff
@@ -56,293 +54,336 @@ use sp_std::{prelude::*, vec};
+			);
 
 			let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
 
 			if Self::is_registered(cid, cindex, &sender) {
-				return Err(<Error<T>>::ParticipantAlreadyRegistered.into());
+				return Err(<Error<T>>::ParticipantAlreadyRegistered.into())
 			}
 
 			if let Some(p) = &proof {
 				// we accept proofs from other communities as well. no need to ensure cid
 				ensure!(sender == p.prover_public, Error::<T>::WrongProofSubject);
 				ensure!(p.ceremony_index < cindex, Error::<T>::ProofAcausal);
-				ensure!(p.ceremony_index >= cindex.checked_sub(T::ReputationLifetime::get()).unwrap_or(0), Error::<T>::ProofOutdated);
-				ensure!(Self::participant_reputation(&(p.community_identifier, p.ceremony_index),
-					&p.attendee_public) == Reputation::VerifiedUnlinked,
-					Error::<T>::AttendanceUnverifiedOrAlreadyUsed);
+				ensure!(
+					p.ceremony_index >=
+						cindex.checked_sub(T::ReputationLifetime::get()).unwrap_or(0),
+					Error::<T>::ProofOutdated
+				);
+				ensure!(
+					Self::participant_reputation(
+						&(p.community_identifier, p.ceremony_index),
+						&p.attendee_public
+					) == Reputation::VerifiedUnlinked,
+					Error::<T>::AttendanceUnverifiedOrAlreadyUsed
+				);
 				if Self::verify_attendee_signature(p.clone()).is_err() {
-					return Err(<Error<T>>::BadProofOfAttendanceSignature.into());
+					return Err(<Error<T>>::BadProofOfAttendanceSignature.into())
 				};
 
 				// this reputation must now be burned so it can not be used again
-				<ParticipantReputation<T>>::insert(&(p.community_identifier, p.ceremony_index),
-					&p.attendee_public, Reputation::VerifiedLinked);
+				<ParticipantReputation<T>>::insert(
+					&(p.community_identifier, p.ceremony_index),
+					&p.attendee_public,
+					Reputation::VerifiedLinked,
+				);
 				// register participant as reputable
-				<ParticipantReputation<T>>::insert((cid, cindex),
-					&sender, Reputation::UnverifiedReputable);
+				<ParticipantReputation<T>>::insert(
+					(cid, cindex),
+					&sender,
+					Reputation::UnverifiedReputable,
+				);
 			};
 
 			Self::register(cid, cindex, &sender, proof.is_some())?;
 
 			debug!(target: LOG, "registered participant: {:?}", sender);
-			Ok(())
+
+			Self::deposit_event(Event::ParticipantRegistered(sender));
+			Ok(().into())
 		}
 
-		#[weight = 10_000]
-		pub fn attest_claims(origin, claims: Vec<ClaimOfAttendance<T::Signature, T::AccountId, T::Moment>>) -> DispatchResult {
+		#[pallet::weight(10_000)]
+		pub fn attest_claims(
+			origin: OriginFor<T>,
+			claims: Vec<ClaimOfAttendance<T::Signature, T::AccountId, T::Moment>>,
+		) -> DispatchResultWithPostInfo {
 			let sender = ensure_signed(origin)?;
-			ensure!(<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::ATTESTING,
-				Error::<T>::AttestationPhaseRequired);
+			ensure!(
+				<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::ATTESTING,
+				Error::<T>::AttestationPhaseRequired
+			);
```

Comment:
A lot of fmt because the rustfmt can now analyze the code in the procedural macro. Before, the analyzer did not dare to touch the code, as it might have bricked something.

Reply:
Ok